collectors/ebpf.plugin/ebpf_apps.c

```diff
@@ -956,19 +885,34 @@ static void cleanup_exited_pids(void) {
     }
 }
 
-int collect_data_for_all_processes(
-    int (*bpf_map_get_next_key)(int, const void *, void *), int (*bpf_map_lookup_elem)(int, const void *, void *),
-    int tbl_pid_stats_fd)
+#ifndef STATIC
+int collect_data_for_all_processes(ebpf_process_stat_t **out,
+                                   pid_t *index,
+                                   int (*bpf_map_get_next_key)(int, const void *, void *),
+                                   int (*bpf_map_lookup_elem)(int, const void *, void *),
+                                   int tbl_pid_stats_fd)
+#else
+int collect_data_for_all_processes(ebpf_process_stat_t **out,
+                                   pid_t *index,
+                                   int tbl_pid_stats_fd)
+#endif
 {
```

Comment:
Why are you doing it this way? Why don't you assign function addresses in `process_functions` depending on `STATIC` macro?

Reply:
The `kickstart_static64` will need a different structure to compile, the `STATIC` is already used on `netdata/kernel-collector` when we need to drop completely the shared libraries, I am keeping this here for while to keep the same pattern used on the other repository.

Reply:
I'm not against using `STATIC`. I just prefer to do it in one place rather than several ones.